source/Calamari.Aws/Kubernetes/Discovery/AwsAuthenticationDetails.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections.Generic;
+using Amazon.Runtime;
+using Calamari.Common.Features.Discovery;
+
+namespace Calamari.Kubernetes.Aws
+{
+    public class AwsAuthenticationDetails : ITargetDiscoveryAuthenticationDetails
+    {
+        const string DefaultSessionName = "OctopusKubernetesClusterDiscovery";
+        
+        /// <exception cref="AggregateException">
+        /// If both InstanceProfile and EnvironmentVariable Credentials fail.
+        /// Contains AmazonClientExceptions for both InstanceProfile and EnvironmentVariable failures</exception>
+        /// <exception cref="AmazonClientException">If Basic (Account) Credentials fail</exception>
+        public AWSCredentials ToCredentials()
+        {
+            AWSCredentials account;
+            if (Credentials.Type == "account")
+            {
+                account = new BasicAWSCredentials(Credentials.Account.AccessKey, Credentials.Account.SecretKey);
+            }
+            else
+            {
+                try
+                {
+                    // If not currently running on an EC2 instance,
+                    // this will throw an exception.
+                    account = new InstanceProfileAWSCredentials();
+                }
+                catch (Exception instanceProfileException)
```

Comment:
We really should be catching a more specific exception here.
It looks like this constructor and `EnvironmentVariablesAWSCredentials` below will both throw an `InvalidOperationException` if they fail, so we should just catch that.

Reply:
As discussed it's not clear what exception we might receive from inspecting the decompiled code so I've added a comment and left the exception handlers catching the base Exception.